src/app/components/harvest/components/shared/title.component.ts

```diff
@@ -1,11 +1,37 @@
 import { Component, Input } from "@angular/core";
+import { NgForm } from "@angular/forms";
+import { ShallowHarvestsService } from "@baw-api/harvest/harvest.service";
+import { BawApiError } from "@helpers/custom-errors/baw-api-error";
+import { Harvest } from "@models/Harvest";
 import { Project } from "@models/Project";
+import { throwError } from "rxjs";
 
 @Component({
   selector: "baw-harvest-title",
   template: `
     <h1>
-      <small class="text-muted"> Project: {{ project.name }} </small>
+      <small class="text-muted">
+        <form #f="ngForm" (ngSubmit)="updateHarvestName(f)">
+          Project: {{ project.name }} | Harvest: {{ !editingHarvestName ? harvest.name : '' }}
+          <input
+            name="harvestNameInput"
+            *ngIf="editingHarvestName"
+            (keyup.escape)="toggleHarvestNameEditing()"
+            [ngModel]="harvest.name"
+            placeholder="{{harvest.name}}"
+            maxlength="22"
+            ngModel
```

Comment:
doesn't follow our design system - needs bootstrap styling

Reply:
Thanks! I've made the ticket changes and will start working on save/cancel buttons